### DISCUSSION TITLE
Azure Devsops Server gittools extension hanging on 'gitversion/execute' - takes 8.5 minutes

### DISCUSSION THREAD

**Discussion by blakeduffey** [GitTools/GitVersion#3581] [Q&A]
Azure Devops Server 2022
GitTools extension installed

Pipeline snippet looks like:

```
# gittools setup for gitsemver versioning
- task: gittools.gittools.setup-gitversion-task.gitversion/setup@0
  displayName: gitversion/setup
  inputs:
    versionSpec: '*'
    preferLatestVersion: true
  
# gittools execute semver
- task: gittools.gittools.execute-gitversion-task.gitversion/execute@0
  displayName: gitversion/execute

I use these steps in other pipelines without issue

I'm using it in a new pipeline and it hangs on the gitversion/execute step for 8 minutes, 29 seconds.  Is this normal/related to the git history?  Is there a way to speed it along?  :)

**Comment by HHobeck** [comment]
I'm moving this to the discussion section because it seems to be not a bug report.

**Comment by blakeduffey** [comment]
So I'll clarify.  I have an Azure Devops Server pipeline.

Using the legacy version extension

`steps:
- task: gittools.gitversion.gitversion-task.GitVersion@5
  displayName: GitVersion
  inputs:
    runtime: full
    configFilePath: GitVersion.yml`

This step takes about two minutes.

Using the current version

`- task: gittools.gittools.setup-gitversion-task.gitversion/setup@0
  displayName: gitversion/setup
  inputs:
    versionSpec: '*'
    preferLatestVersion: true
    
- task: gittools.gittools.execute-gitversion-task.gitversion/execute@0
  displayName: gitversion/execute`

The same source code/git history takes nearly 10 minutes.

Would appreciate any feedback that might point me in the right direction.

**Reply by HHobeck** [reply]
Hmm good question. @arturcic do you have an explanation for that behavior? Maybe it is solved in the pre-release version of git version!?

https://github.com/GitTools/GitVersion/releases/tag/6.0.0-beta.2
